### PR TITLE
Add missing Changelogs for October releases

### PR DESCRIPTION
**Pull request by tmattio**
cc @kit-ty-kate @panglesd I'm adding changelogs for the latest releases of odoc and opam-publish. Would be very appreciated if you added the entry to the [OCaml Changelog](https://ocaml.org/changelog) for the next releases of Platform tools 🙂

### CODE REVIEW

**Review by christinerose — APPROVED** [review]

File: data/changelog/odoc/2023-10-02-odoc-2.3.1.md [hunk 1]
```diff
@@ -1,9 +1,9 @@
 ---
-title: Odoc 2.3.1
+title: `odoc` 2.3.1
```
Comment (tmattio) [inline data/changelog/odoc/2023-10-02-odoc-2.3.1.md]: @christinerose Isn't this reserved for the CLI tool, and Odoc is the name of the projects?

Comment (christinerose) [reply data/changelog/odoc/2023-10-02-odoc-2.3.1.md]: According to Jon, `odoc` is the proper name of the project as well as the CLI tool, unless he's changed his mind.

**Comment by tmattio** [comment]
@christinerose I'm reverting some of the formatting changes since they broke the build. I also think it'd be better to make the changes on all the changelog entries so there's consistency in the formatting when reading the page.

**Comment by christinerose** [comment]
> make the changes on all the changelog entries so there's consistency in the formatting when reading the page

Agreed!

**Pull request merged**